administration/src/bp-modules/util/cardGenerationError.tsx

```diff
@@ -0,0 +1,43 @@
+import { ApolloError } from '@apollo/client'
+import { Toaster } from '@blueprintjs/core'
+import { TFunction } from 'i18next'
+import React from 'react'
+
+import { CsvError } from '../../cards/CsvFactory'
+import { PdfError } from '../../cards/PdfFactory'
+import { CreateCardsError } from '../../cards/createCards'
+import getMessageFromApolloError from '../../errors/getMessageFromApolloError'
+import { reportErrorToSentry } from '../../util/sentry'
+import FormErrorMessage from '../self-service/components/FormErrorMessage'
+
+export const showCardGenerationError = (appToaster: Toaster, error: unknown, t: TFunction<'errors'>): void => {
+  if (error instanceof CreateCardsError) {
+    appToaster.show({
+      message: error.message,
+      intent: 'danger',
+    })
+  } else if (error instanceof ApolloError) {
+    const { title } = getMessageFromApolloError(error, t)
+    appToaster.show({
+      message: <FormErrorMessage style={{ color: 'white' }} errorMessage={title} />,
+      timeout: 0,
+      intent: 'danger',
+    })
+  } else if (error instanceof PdfError) {
+    appToaster.show({
+      message: t('pdfCreationError'),
```

Comment:
🔧  just a general suggestion.
I think we could avoid passing all the `TFunction` by using `i18next.t('errors:csvCreationError')`
I also used to pass the `t` down to everywhere and the amount of function parameters just increase.

Reply:
I think this was enough refactoring in this PR and not done by me, therefore I would really not do this here anymore :thinking: However, I planned to do a follow up PR anyway to address this issue. I would do it in a different way though by only returning the key without translating it any only translating it where it is actually shown to the user.